src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java

```diff
@@ -573,6 +578,35 @@ private void batchStoreDirectoryEntry(
         }
     }
 
+    // We make a special method just for validating id, because when
+    // validating we do not need to actually get the data.
+    private List<String> batchValidateIds(
+        Firestore firestore, String collectionId, List<String> batch)
+        throws InterruptedException {
+        CollectionReference datasetCollection = firestore.collection(collectionId);
+
+        List<String> missingIds = new ArrayList<>();
+        List<ApiFuture<QuerySnapshot>> futures = new ArrayList<>();
+        for (String fileId : batch) {
+            Query query = datasetCollection.whereEqualTo("fileId", fileId);
+            futures.add(query.get());
```

Comment:
Question: It looks like many of our other query calls are wrapped in a try/catch or are called from a transaction. Is there a reason why we're not doing that here?

Reply:
any errors from the query are thrown when we get the results of the query and that happens inside of the try catch.
I could extend the tribe but nothing above there throws ExecutionException.

Reply:
Gotcha, makes sense!